pkg/apis/metal/v1alpha1/types_cloudprofile.go

```diff
@@ -72,4 +72,50 @@ const (
 type Partition struct {
 	// FirewallTypes is a list of available firewall machine types in this partition. When empty, allows all values.
 	FirewallTypes []string `json:"firewallTypes"`
+
+	// NetworkIsolation if given allows the creation of shoot clusters which have network restrictions activated.
+	NetworkIsolation *NetworkIsolation `json:"networkIsolation,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageProviderConfig is stored in gardenv1beta1.Worker.Image.ProviderConfig as RawExtension
+// to configure metal specific worker configuration items
+type ImageProviderConfig struct {
```

Comment:
Should be moved to `types_worker.go`? Be aware that the shoot resource is a user-facing API. Probably it is worth to document that this gets defaulted by the provider and is not supposed to be set by the user. Maybe even disallow this to be set by the user through a validation and just encode these configurations on the fly.

Reply:
If you plan to do it this way, the shoot secret mutation also needs adaption such that the network isolation configuration must be taken from the cloud profile and not from the worker configuration.